File: C++/2033_MinimumOperations.cpp
class Solution {
public:
    int minOperations(vector<vector<int>>& grid, int x) {
        vector<int> a;
        for(int i = 0; i < grid.size();i++){
            for(int j = 0; j < grid[i].size(); j++){
                a.push_back(grid[i][j]);
            }
        }
        sort(a.begin(), a.end());
        int median = a[a.size()/2];
        int output = 0;
        for (int b : a){
            if (abs(b-median)%x == 0){
                output += abs(b-median)/x;
            } else{
                return -1;
            }
        }
        return output;
    }
};

File: C#/2033_MinimumOperations.cs
public class Solution {
    public int MinOperations(int[][] grid, int x) {
        List<int> a = new List<int>();
        for(int i = 0; i < grid.Length; i++){
            for(int j = 0; j < grid[i].Length; j++){
                a.Add(grid[i][j]);
            }
        }
        a.Sort();
        int median = a[a.Count/2];
        int output = 0;
        for(int z = 0; z < a.Count; z++){
            if (Math.Abs(median-a[z])%x == 0){
                output += Math.Abs(median-a[z])/x;
            } else{
                return -1;
            }
        }
        return output;
    }
}